src/utils/sformat.fs

```diff
@@ -323,7 +323,15 @@ namespace Microsoft.FSharp.Text.StructuredFormat
                 // to 7.
 
                 if FSharpType.IsTuple reprty then 
-                    TupleValue (FSharpValue.GetTupleFields obj |> Array.toList)
+                    let tyArgs = reprty.GetGenericArguments()
```

Comment:
@dsyme  I'm getting error when using it
![image](https://cloud.githubusercontent.com/assets/1097246/20499299/97d70ff0-b039-11e6-8b96-db359986a1ce.png)